### ISSUE TITLE
Checkbox Rect should contain all text inside it

### ISSUE THREAD

**Issue by AaronM04** [conwayste/conwayste#89]
Currently the Checkbox's label will overflow the Rect. The text should be cropped to fit inside the Rect. In no cases should anything be drawn outside of it. The checkbox itself should always be the largest square that fits on the left end of the Rect.

Also, it would be nice to have a `shrink_to_fit` method that would shrink the Rect to perfectly fit the checkbox+label combination. Parent widgets (like panes) could then call this and adjust the checkbox's position and/or their own size to neatly fit everything.

**Comment by AaronM04** [comment]
What do you think @Manghi -- does this belong in 1.0 milestone? And does it deserve an easy label? :)

**Comment by Manghi** [comment]
Yes it does. Sorry this slipped my radar. As we discussed offline, the rectangle should either grow-to-fit or shrink-to-fit depending on Label's dimensions. We had also talked about multi-line support, but we can save that for another ticket.

**Comment by AaronM04** [comment]
Actually I'm not 100% sure we want the label inside the checkbox widget.

**Comment by Manghi** [comment]
That makes a lot more sense to be honest. The case I was covering was being able to click on the label and have the checkbox toggle as well -- this is a pretty common feature of checkboxes across several platforms. I think with what you described, the overall "grid" would receive on click event and then process the state of its child, the checkbox.

**Comment by Manghi** [comment]
Should be doable after #94 is merged
